query-engine/connectors/query-connector/src/filter/json.rs

```diff
@@ -27,7 +25,7 @@ impl JsonCompare for ScalarFieldRef {
                 path,
                 target_type: Some(target_type),
             }),
-            projection: ScalarProjection::Single(Arc::clone(self)),
+            projection: ScalarProjection::Single(self.clone()),
```

Comment:
Isn't this against our coding guidelines of using explicit `Arc::clone()` everywhere instead of `.clone()` :trollface:

Reply:
The problem is that it's a type error now, because ScalarFieldRef is a Zipper (that contains an Arc), not an Arc. We could use `Zipper::clone()` but that feels a bit weird.